src/cc65/coptmisc.c

```diff
@@ -555,6 +556,65 @@ unsigned OptAXLoad (CodeSeg* S)
 
 
 
+unsigned OptAXLoad2 (CodeSeg* S)
+/* Merge ldy/jsr incaxy/jsr ldaxi into ldy/jsr ldaxidx */
+{
+    unsigned Changes = 0;
+    unsigned I;
+
+    /* Walk over the entries */
+    I = 0;
+    while (I < CS_GetEntryCount (S)) {
+
+        signed Val;
+        CodeEntry* E[3];
+        CodeEntry *X;
+        char *End;
+
+        /* Get the next entry */
+        E[0] = CS_GetEntry (S, I);
+
+        /* Check for ldy followed by incaxy followed by jsr/jmp ldaxi */
+        if (E[0]->OPC == OP65_LDY                            &&
+            E[0]->AM == AM65_IMM                             &&
+            CS_GetEntries (S, E+1, I+1, 2)                   &&
+            E[1]->OPC == OP65_JSR                            &&
+            strcmp (E[1]->Arg, "incaxy") == 0                &&
+            (E[2]->OPC == OP65_JSR || E[2]->OPC == OP65_JMP) &&
+            strcmp (E[2]->Arg, "ldaxi") == 0                 &&
+            !CS_RangeHasLabel (S, I, 3)) {
+
+            /* Replace with ldy (y+1) / jsr ldaxidx */
+            Val = strtoul(E[0]->Arg + 1, &End, 16);
+            Val++;
```

Comment:
```suggestion
            Val = E[0]->Num + 1;
```
If you use `CE_IsKnownImm()` above, you can just use the - already known - numeric value here.